ultravox/data/datasets.py

```diff
@@ -492,78 +505,95 @@ def _get_sample(self, row: transformers.BatchFeature) -> VoiceSample:
         return self._get_transcribe_sample(row, tcol="question")
 
 
-class BoolQWithExtendedAnswerDataset(BoolQDataset):
+class QAVoiceDatasetMixin(VoiceDataset):
     SEPARATORS = ["\n\n", "\n", "\n----\n"]
-    BOOLQ_PASSAGE_PROMPTS = [
-        "Provide a short explanation, then respond with True/False on the last line",
-        "Explain briefly, concluding with True/False on a new line."
-        "Write a quick explanation, and finish with True/False on the last line"
-        "Summarize in a few words, and end with True/False on a new line."
-        "Give a brief explanation first, then answer with True/False on the final line",
-        "Start with a concise explanation, and end with a True/False response on the last line.",
-        "Explain briefly and follow up with True/False at the end",
-        "Write a short explanation, then state True/False on a new line.",
-        "First, offer a brief explanation, and then reply with True/False at the end.",
-        "Present a concise explanation, ending with a True/False answer on the final line",
-        "Start with a brief explanation, and then answer with True/False at the end.",
-    ]
-    QUERY_PROMPTS = ["Question: ", "Question:\n", "Q: ", "Q:\n", "Query: ", "Query:\n"]
-    CONTEXT_PROMPTS = [
+    QUERY_PREFIX = ["Question: ", "Question:\n", "Q: ", "Q:\n", "Query: ", "Query:\n"]
+    CONTEXT_PREFIX = [
         "Passage: ",
         "Passage:\n",
         "Context: ",
         "Context:\n",
         "Background: ",
         "Background:\n",
     ]
-    ANSWER_PROMPTS = [
+    ANSWER_PREFIX = [
         "Answer: ",
         "A: ",
         "",
         "The answer is: ",
         "Result: ",
         "Conclusion: ",
     ]
+    # In most cases there is no extra prompt-suffix needed
+    PROMPT_SUFFIXES = [""]
 
-    def _get_query_prompt(self) -> str:
+    def _get_query_prompt(self, question_str: str, context: str) -> str:
```

Comment:
this ends up overlapping with the functionality in `_get_answer_messages`. Should we remove `_get_answer_prompt` and `_get_answer_messages` and just use this mixin for all non-transcribe datasets?

Reply:
Okay so the main issue I have for combining them is that `ANSWER_PROMPTS` don't assume full audio-text equivalence. For example, we have the prompt `Listen to <|audio|> and respond to it`.

Now, what is the text-equivalent of that exactly? In `_get_answer_messages` we're saying the whole prompt (with the audio in it) is equivalent to the question's text. Maybe that's okay there (I've never liked that), but when trying to combine them things get a bit weird. Also, what is the role of `self._args.prompt` at that point? Does it go at the start or at the end?

I propose to get rid of `ANSWER_PROMPTS` and essentially only have `<|audio|>` there. I will instead embellish `QUERY_PREFIX = ["Question: ", "Question:\n", "Q: ", "Q:\n", "Query: ", "Query:\n"]` with variations like `ANSWER_PROMPT`, but make them agnostic to the modality of the input. Thoughts?

Reply:
Since this won't be easily verifiable, the prudent thing to do is to train with it and test the resulting model. That would probably require a new PR though.

Reply:
I think if we want multi-task behavior, we need text instructions. But I agree we should make the result agnostic of modality as you describe. For now just maybe leave a comment with the plan here and we can revisit it in a future PR.

Reply:
Done.